### PR TITLE
Serialization/deserialization of key and micro blocks

### PR DESCRIPTION
**Pull request by jur0**

### CODE REVIEW

File: apps/aecore/src/aec_blocks.erl [hunk 1]
```diff
@@ -162,133 +169,208 @@ new_key_with_state(LastBlock, CurrentKeyBlock, Miner, Trees0) ->
...
+    #header{height = Height,
+            prev_hash = PrevHash,
+            root_hash = RootHash,
+            txs_hash = TxsHash,
+            miner_hash = MinerHash,
+            signature = Signature,
+            time = Time,
+            version = Version}.
+
+from_header_and_txs(#header{} = H, Txs) ->
```
Comment (hanssv) [inline apps/aecore/src/aec_blocks.erl]: This looks strange both `key` and `micro` type ignores the Txs? What would then be the point of this function?

File: apps/aecore/src/aec_blocks.erl [hunk 2]
```diff
@@ -162,133 +169,208 @@ new_key_with_state(LastBlock, CurrentKeyBlock, Miner, Trees0) ->
     NewHeight = LastBlockHeight + 1,
     Version = protocol_effective_at_height(NewHeight),
 
-    {UncompleteBlock#block{height = NewHeight, version = Version, miner = KeyToClaimLeader, key_hash = undefined}, Trees}.
+    {UncompleteBlock#block{height = NewHeight, version = Version, miner = KeyToClaimLeader, miner_hash = undefined}, Trees}.
```
Comment (hanssv) [inline apps/aecore/src/aec_blocks.erl]: Bad English: s/Uncomplete/Incomplete

File: apps/aecore/src/aec_blocks.erl (hunk 1 above)
Comment (jur0) [reply apps/aecore/src/aec_blocks.erl]: This is still WIP, `Txs` won't be ignored in microblocks.

**Comment by sennui** [comment]
lets merge after green tests in ng-develop (and rebasing it on ng-develop)

**Comment by lucafavatella** [comment]
I take liberty of labelling as WIP in order to ease filtering for release.

**Comment by jur0** [comment]
Closing as it's already in the current NG branch #1260.

**Pull request closed without merging**